pkg/preparer/orchestrate.go

```diff
@@ -150,22 +151,28 @@ func (p *Preparer) handlePods(podChan <-chan pods.Manifest, quit <-chan struct{}
 			if !working {
 				p.tryRunHooks(hooks.AFTER_AUTH_FAIL, pods.NewPod(manifestToLaunch.ID(), pods.PodPath(p.podRoot, manifestToLaunch.ID())), &manifestToLaunch, manifestLogger)
 			}
-		case <-time.After(1 * time.Second):
+		default:
 			if working {
 				pod := pods.NewPod(manifestToLaunch.ID(), pods.PodPath(p.podRoot, manifestToLaunch.ID()))
 
 				// HACK ZONE. When we have better authz, rewrite.
-				// Still need to ensure that preparer launches correctly
-				// as root
-				if pod.Id == POD_ID {
-					pod.RunAs = "root"
-				}
-
-				ok := p.installAndLaunchPod(&manifestToLaunch, pod, manifestLogger)
-				if ok {
-					manifestToLaunch = pods.Manifest{}
+				if (manifestToLaunch.RunAsUser() == "root") && (pod.Id != POD_ID) {
```

Comment:
I would support a refactor to move the authz logic into installAndLaunchPod. This function is difficult to test (as we're finding out here...) so we should minimize the amount of actual work it does.

I'm also not convinced that the sleep is semantically equivalent to selecting on a timed channel. Was that change necessary for testability?